nemo/collections/asr/parts/utils/diarization_utils.py

```diff
@@ -15,500 +15,194 @@
-        mid_delay = math.ceil(
-            (self.chunk_len_in_sec + (self.total_buffer_in_secs - self.chunk_len_in_sec) / 2)
-            / self.model_stride_in_secs
-        )
-
-        wer_ts = WERBPE_TS(
-            tokenizer=_asr_model.tokenizer,
-            batch_dim_index=0,
-            use_cer=_asr_model._cfg.get('use_cer', False),
-            ctc_decode=True,
-            dist_sync_on_step=True,
-            log_prediction=_asr_model._cfg.get("log_prediction", False),
+    def load_realigning_LM(self):
+        self.N_range = (
+            self.realigning_lm_params['min_number_of_words'],
+            self.realigning_lm_params['max_number_of_words'],
         )
-        frame_asr = FrameBatchASR_Logits(
-            asr_model=_asr_model,
-            frame_len=self.chunk_len_in_sec,
-            total_buffer=self.total_buffer_in_secs,
-            batch_size=self.asr_batch_size,
-        )
-
-        with torch.cuda.amp.autocast():
-            hyps, tokens_list, sample_list = get_wer_feat_logit(
-                self.audio_file_list,
-                frame_asr,
-                self.chunk_len_in_sec,
-                tokens_per_chunk,
-                mid_delay,
-                self.model_stride_in_secs,
-                _asr_model.device,
-            )
-
-            for k, greedy_predictions_list in enumerate(tokens_list):
-                logits_len = torch.from_numpy(np.array([len(greedy_predictions_list)]))
-                greedy_predictions_list = greedy_predictions_list[onset_delay:-mid_delay]
-                greedy_predictions = torch.from_numpy(np.array(greedy_predictions_list)).unsqueeze(0)
-                text, char_ts, word_ts = wer_ts.ctc_decoder_predictions_tensor_with_ts(
-                    self.model_stride_in_secs, greedy_predictions, predictions_len=logits_len
-                )
-                _trans_words, word_ts = text[0].split(), word_ts[0]
-                words_list.append(_trans_words)
-                word_ts_list.append(word_ts)
-                assert len(_trans_words) == len(word_ts)
-        return words_list, word_ts_list
-
-    def apply_delay(self, word_ts):
-        """
-        Compensate the constant delay in the decoder output.
-
-        Arg:
-            word_ts (list): List contains the timestamps of the word sequences.
-
-        Return:
-            word_ts (list): List of the delay-applied word-timestamp values.
-        """
-        for p in range(len(word_ts)):
-            word_ts[p] = [
-                max(round(word_ts[p][0] - self.asr_delay_sec, 2), 0),
-                round(word_ts[p][1] - self.asr_delay_sec, 2),
-            ]
-        return word_ts
+        self.stt_end_tokens = ['</s>', '<s>']
+        if 'arpa' in sys.modules:
+            return arpa.loadf(self.realigning_lm_params['arpa_language_model'])[0]
+        else:
+            logging.info("You must install arpa to refine diarization result with LM. Skipping realigning.")
```

Comment:
Throw error here if user requested for realigning with LM but arpa or KenLM not present. with comments something like 
`you requested to use KenLM LM but its not installed. Please install from <LINK> and try again or run without realining LM case`

Reply:
Fixed as discussed.